inginious/common/hook_manager.py

```diff
@@ -31,3 +31,13 @@ def add_hook(self, name, callback):
     def call_hook(self, name, **kwargs):
         """ Call all hooks registered with this name. Returns a list of the returns values of the hooks (in the order the hooks were added)"""
         return [y for y in [x(**kwargs) for x in self.hooks.get(name, [])] if y is not None]
+
+    def call_hook_recursive(self, name, **kwargs):
```

Comment:
It looks more like a piping system than a recursive call...am I wrong ? Nice, but we should be careful on the way hooks are sequenced here...hooks must be independent.